File: warrior/systems/MovementSystem.java
package net.brokenspork.systems;

import net.brokenspork.components.Position;
import net.brokenspork.components.Velocity;

import com.artemis.Aspect;
import com.artemis.ComponentMapper;
import com.artemis.Entity;
import com.artemis.annotations.Mapper;
import com.artemis.systems.EntityProcessingSystem;

public class MovementSystem extends EntityProcessingSystem {
	@Mapper ComponentMapper<Position> pm;
	@Mapper ComponentMapper<Velocity> vm;

	@SuppressWarnings("unchecked")
    public MovementSystem() {
		super(Aspect.getAspectForAll(Position.class, Velocity.class));
	}

	@Override
	protected void process(Entity e) {
		Position position = pm.get(e);
		Velocity velocity = vm.get(e);
		
		if(velocity == null) {
		    return;
		}
		position.x += velocity.vectorX*world.delta;
		position.y += velocity.vectorY*world.delta;
	}

}

File: example/systems/MovementSystem.ts
module brokenspork.systems {
	
	import Position = brokenspork.components.Position;
	import Velocity = brokenspork.components.Velocity;
	
	import Aspect = artemis.Aspect;
	import ComponentMapper = artemis.ComponentMapper;
	import Entity = artemis.Entity;
	import EntityProcessingSystem = artemis.systems.EntityProcessingSystem;
	
	export class MovementSystem extends EntityProcessingSystem {
		@Mapper Position
		pm:ComponentMapper<Position>;
		@Mapper Velocity
		vm:ComponentMapper<Velocity>;
	
		//@SuppressWarnings("unchecked")
		constructor() {
			super(Aspect.getAspectForAll(Position, Velocity));
		}
	
		
		public process(e:Entity) {
			var position:Position = this.pm.get(e);
			var velocity:Velocity = this.vm.get(e);
			
			if(velocity == null) {
					return;
			}
			position.x += velocity.vectorX*this.world.delta;
			position.y += velocity.vectorY*this.world.delta;
		}
	
	}
}

